### PR TITLE
Added phone number search

### PR DESCRIPTION
**Pull request by k4u5h1k**
Used numverify.com to implement basic phone number enumeration.

Their free api only allows 250 requests and their gui uses AJAX with a weak verification hash
which is made using,  `md5(input_phone_number+secret_tag_value)`

Simply forging this and making requests to their internal api to get unlimited requests.

Even works for landline numbers!

Screenshot 2021-04-03 at 11 30 53 PM

### CODE REVIEW

File: modules/osint/phone_number_search.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        'sources': ('numverify'),
+        'options': (
+                ('number', None, True, 'Phone number (Must include area code)', '-n', 'store', str),
+        ),
+        'examples': ('phone_number_search -q 911234567890')
+}
+
+def module_api(self):
+	num = self.options['number']
+	if re.search(r"^\+\d{1,2}\s?\d{10}$", num):
```
Comment (saeeddhqan) [inline modules/osint/phone_number_search.py]: remove if-else structure. If the phone is invalid, the API can handle it.

Comment (k4u5h1k) [reply modules/osint/phone_number_search.py:33]: You mean i should move it to the engine?

Comment (k4u5h1k) [reply modules/osint/phone_number_search.py]: How do I tell the user the correct format?
+{area code}{ten digit number}

Comment (saeeddhqan) [reply modules/osint/phone_number_search.py]: How the API tell you that the input phone is invalid? also any alert and print in module_api won't print.

Comment (k4u5h1k) [reply modules/osint/phone_number_search.py]: Just a ten digit number without area code is invalid, but it doesn't say what is wrong, so I have to.

Output after removing number check
Screenshot 2021-04-04 at 12 07 41 AM

oh you mean I just check whether output['valid'] or not in module_run and print accordingly

Comment (k4u5h1k) [reply modules/osint/phone_number_search.py:33]: self.error works just fine from inside module_api
Screenshot 2021-04-04 at 12 48 46 AM

**Pull request merged**